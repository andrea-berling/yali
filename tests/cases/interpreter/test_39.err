[line 17] Error at 'world': Undefined reference.
