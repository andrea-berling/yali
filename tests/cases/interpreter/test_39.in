// This program tests that when a variable is declared in an outer scope,
// it can be used in an inner scope
// But when a variable is declared in an inner scope,
// it can't be used in an outer scope
{
  var world = "outer world";
  var quz = "outer quz";
  {
    world = "modified world";
    var quz = "inner quz";
    print world;
    print quz;
  }
  print world;
  print quz;
}
print world;
