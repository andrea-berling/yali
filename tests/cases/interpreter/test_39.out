modified world
inner quz
modified world
outer quz
